### PR TITLE
Makefile: Remove tmpdir after container build

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:
If you run `make container` multiple times, you'll have a bunch of temporary directories on your machine. Additionally added some style fixes to make it more uniform.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,10 +91,11 @@ all-push: $(addprefix sub-push-,$(ALL_ARCH))
 
 container: .container-$(ARCH)
 .container-$(ARCH):
-	docker run --rm -v "$$PWD":/go/src/k8s.io/kube-state-metrics -w /go/src/k8s.io/kube-state-metrics -e GOOS=linux -e GOARCH=$(ARCH) -e CGO_ENABLED=0 golang:${GO_VERSION} go build -ldflags "-s -w -X ${PKG}/version.Release=${TAG} -X ${PKG}/version.Commit=${Commit} -X ${PKG}/version.BuildDate=${BuildDate}" -o kube-state-metrics
-	cp -r * $(TEMP_DIR)
-	docker build -t $(MULTI_ARCH_IMG):$(TAG) $(TEMP_DIR)
+	docker run --rm -v "${PWD}:/go/src/k8s.io/kube-state-metrics" -w /go/src/k8s.io/kube-state-metrics -e GOOS=linux -e GOARCH=$(ARCH) -e CGO_ENABLED=0 golang:${GO_VERSION} go build -ldflags "-s -w -X ${PKG}/version.Release=${TAG} -X ${PKG}/version.Commit=${Commit} -X ${PKG}/version.BuildDate=${BuildDate}" -o kube-state-metrics
+	cp -r * "$(TEMP_DIR)"
```
Comment (tariq1890) [inline Makefile]:
```suggestion
	cp -r * "${TEMP_DIR}"
```

File: Makefile [hunk 2]
```diff
@@ -91,10 +91,11 @@ all-push: $(addprefix sub-push-,$(ALL_ARCH))
...
+	docker build -t $(MULTI_ARCH_IMG):$(TAG) "$(TEMP_DIR)"
```
Comment (tariq1890) [inline Makefile]:
```suggestion
	docker build -t $(MULTI_ARCH_IMG):$(TAG) "${TEMP_DIR}"
```

File: Makefile [hunk 3]
```diff
@@ -91,10 +91,11 @@ all-push: $(addprefix sub-push-,$(ALL_ARCH))
...
 	docker tag $(MULTI_ARCH_IMG):$(TAG) $(MULTI_ARCH_IMG):latest
+	rm -rf "$(TEMP_DIR)"
```
Comment (tariq1890) [inline Makefile]:
```suggestion
	rm -rf "${TEMP_DIR}"
```

**Comment by mrueg** [comment]
Thanks, applied the suggestions and rebased.

**Pull request merged**